zenoh/src/api/publisher.rs

```diff
@@ -990,15 +830,20 @@ impl<'a> MatchingListenerInner<'a> {
 }
 
 #[zenoh_macros::unstable]
-impl<'a> UndeclarableSealed<(), MatchingListenerUndeclaration<'a>> for MatchingListenerInner<'a> {
-    fn undeclare_inner(self, _: ()) -> MatchingListenerUndeclaration<'a> {
+impl<'a> UndeclarableSealed<()> for MatchingListenerInner<'a> {
+    type Undeclaration = MatchingListenerUndeclaration<'a>;
+
+    fn undeclare_inner(self, _: ()) -> Self::Undeclaration {
         MatchingListenerUndeclaration { subscriber: self }
     }
 }
 
 /// A listener that sends notifications when the [`MatchingStatus`] of a
 /// publisher changes.
 ///
+/// Matching litsteners run in background until the publisher is undeclared.
+/// They can be manually undeclared, but will not be undeclared on drop.
```

Comment:
Does `They` refer to both `matching listener` and `publisher`? What does `but will not be undeclared on drop` refer to here?

Reply:
It's seems more like remains of my previous works. `MatchingListener` should follow the same semantic than `Subscribers`, i.e. undeclaration on drop if the handler is not a ZST. I will fix it.